sklearn/preprocessing/_csr_polynomial_expansion.pyx

```diff
@@ -1,20 +1,46 @@
-# Author: Andrew nystrom <awnystrom@gmail.com>
-
+# Authors: Andrew nystrom <awnystrom@gmail.com>
+#          Meekail Zain <zainmeekail@gmail.com>
 from scipy.sparse import csr_matrix
-cimport numpy as cnp
+from libc.math cimport sqrt, pow
 
+cimport numpy as cnp
 cnp.import_array()
-ctypedef cnp.int32_t INDEX_T
 
-ctypedef fused DATA_T:
+cdef extern from "limits.h":
+    cdef int INT_MAX
+    cdef long LONG_MAX
```

Comment:
Based [the eponymous pxd file for Cython `0.29.x`](https://github.com/cython/cython/blob/0.29.x/Cython/Includes/libc/limits.pxd), those constants should be `cimport`-able from it:
```suggestion
from lib.limits cimport INT_MAX, LONG_MAX
```